industrial_ci/src/tests/source_tests.sh

```diff
@@ -26,6 +26,25 @@ function install_catkin_lint {
     ici_install_pypi_pkgs_for_command catkin_lint "catkin-lint"
 }
 
+function clang_tidy_filter {
+    local build=$1
+    local base_ref=$2
+    local result
+    local src_dir
+
+    src_dir=$(grep "^CMAKE_HOME_DIRECTORY:INTERNAL=" "$build/CMakeCache.txt")
+    pushd "$(realpath "${src_dir#*=}")" > /dev/null || exit
+    if [ "$(git rev-parse --abbrev-ref HEAD)" == "$base_ref" ] ; then
+        result=1  # HEAD is on par with $base_ref: always perform clang-tidy
+    else
+        git fetch -q origin "$base_ref"
+        # Perform clang-tidy if there were any changed files
+        result=$(git diff --name-only --diff-filter=MA FETCH_HEAD..HEAD . | grep -cP "\.(c|cpp|h|hpp)$")
```

Comment:
Nit: I use single quotes for regexes whenever possible, since they use too many nutty characters.

Reply:
done